genesis/sensors/sensor_manager.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any, Type, TYPE_CHECKING
+
+import torch
+
+from genesis.utils.ring_buffer import TensorRingBuffer
+
+if TYPE_CHECKING:
+    from genesis.options.sensors import SensorOptions
+    from .base_sensor import Sensor
+
+
+class SensorManager:
+    SENSOR_TYPES_MAP: dict[Type["SensorOptions"], Type["Sensor"]] = {}
+
+    def __init__(self, sim):
+        self._sim = sim
+        self._sensors_by_type: dict[Type["Sensor"], list["Sensor"]] = {}
+        self._sensors_metadata: dict[Type["Sensor"], dict[str, Any]] = {}
+        self._gt_cache: dict[Type["Sensor"], torch.Tensor] = {}
```

Comment:
For cache could be global, and sensor-type based. it is the big advantage of storing data flattened. But you need one cache per datatype still.